src/passes/MemoryPacking.cpp

```diff
@@ -120,6 +123,19 @@ struct MemoryPacking : public Pass {
     }
     module->memory.segments.swap(packed);
   }
+
+  void optimizeTrappingBulkMemoryOps(Module* module) {
+    struct Trapper : PostWalker<Trapper> {
+      void process(Expression* curr, Index index) {
+        if (!getModule()->memory.segments[index].isPassive) {
+          ExpressionManipulator::unreachable(curr);
+        }
```

Comment:
Isn't it already a validation error to call memory.init or memory.drop on active segments?

Reply:
Nope, for some reason that is valid but traps (because the segments were dropped during initialization).

Reply:
Oh I see.  I guess that is because validation is done with only the information in the data count section which is not enough to know about the active/passive status of as segment.  Otherwise it would make sense to be a validation error.

Why would we ever expect to have such instruction in the input?  Perhaps binaryen could make this a hard error rather than delaying the error until runtime?

Reply:
Yeah I briefly discussed the possibility of making this a binaryen validation error with @kripken, but I don't believe we have precedent for Binaryen rejecting valid wasm modules, even if they do silly things. We decided this would be better because it still accepts all valid modules and is a nice size optimization, too.

Reply:
I opened https://github.com/WebAssembly/bulk-memory-operations/issues/107 to get some clarifation, but I guess its too late to change DataCount section now.